app.js

```diff
@@ -18,33 +18,32 @@ app.use(async function(ctx, next) {
             ctx.throw(e.response.status, e.response.text);
         }
 
-        if (e instanceof httpErrors.Forbidden) {
+        switch (e.status) {
+        case 400:
+        case 403:
+        case 404:
             ctx.throw(e);
+            break;
+        default:
+            // TODO: Figure out which errors should be exposed to user
+            console.error('Error: ', e, JSON.stringify(e));
+            ctx.throw(400, e.toString());
         }
-
-        // TODO: Figure out which errors should be exposed to user
-        console.error('Error: ', e, JSON.stringify(e));
-        ctx.throw(400, e.toString());
     }
 });
 
 const Router = require('koa-router');
 const router = new Router();
 
 const creatorKeyJson = JSON.parse(process.env.ACCOUNT_CREATOR_KEY);
-const recoveryKeyJson = JSON.parse(process.env.ACCOUNT_RECOVERY_KEY);
 const keyStore = {
-    async getKey(networkId, accountId) {
-        if (accountId == creatorKeyJson.account_id) {
-            return KeyPair.fromString(creatorKeyJson.secret_key || creatorKeyJson.private_key);
-        }
-        // For account recovery purposes use recovery key when updating any account
-        return KeyPair.fromString(recoveryKeyJson.secret_key || creatorKeyJson.private_key);
+    async getKey() {
```

Comment:
I think at this point it makes sense to use stock `InMemoryKeyStore` and just add key from `creatorKeyJson` there. Custom logic was only needed for `recoveryKeyJson`

Reply:
I tried that and though the tests passed locally, the app did not work in a deployed environment. I was getting an "unknown public key" error, iirc. See commit 22862aa for the `InMemoryKeyStore` implementation that I removed.

Reply:
@chadoh I think we should better debug this (e.g. by logging what gets passed to `getKey` as arguments). Something else might be going wrong if wrong account is used to do stuff (that would explain public key not being found).

Reply:
Here's the version that didn't work:

```js
const creatorKeyJson = JSON.parse(process.env.ACCOUNT_CREATOR_KEY);
const keyStore = new nearlib.keyStores.InMemoryKeyStore();
const keyPair = nearlib.KeyPair.fromString(creatorKeyJson.secret_key || creatorKeyJson.private_key);
keyStore.setKey(process.env.NETWORK_ID, creatorKeyJson.account_id, keyPair);

const nearPromise = (async () => {
    const near = await nearlib.connect({
        deps: { keyStore },
        masterAccount: creatorKeyJson.account_id,
        nodeUrl: process.env.NODE_URL
    });
    return near;
})();
```

If I remember correctly, the error involved the `masterAccount: creatorKeyJson.account_id` line. It seemed like maybe the `keyPair = nearlib.KeyPair.fromString(...)` worked properly, but the `keyStore.setKey` didn't. I wondered if `keyStore.setKey` was running on app startup, but not re-running on every individual request to the app.

I doubt it's worth troubleshooting this right now. There's a lot of other stuff here that I want to land, and the current implementation works.